connectors/aws/aws-textract/src/main/java/io/camunda/connector/textract/TextractConnectorFunction.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ *       under one or more contributor license agreements. Licensed under a proprietary license.
+ *       See the License.txt file for more information. You may not use this file
+ *       except in compliance with the proprietary license.
+ */
+package io.camunda.connector.textract;
+
+import io.camunda.connector.api.annotation.OutboundConnector;
+import io.camunda.connector.api.outbound.OutboundConnectorContext;
+import io.camunda.connector.api.outbound.OutboundConnectorFunction;
+import io.camunda.connector.generator.java.annotation.ElementTemplate;
+import io.camunda.connector.textract.caller.AsyncTextractCaller;
+import io.camunda.connector.textract.caller.PollingTextractCalller;
+import io.camunda.connector.textract.caller.SyncTextractCaller;
+import io.camunda.connector.textract.model.TextractRequest;
+import io.camunda.connector.textract.suppliers.AmazonTextractClientSupplier;
+
+@OutboundConnector(
+    name = "AWS Textract",
+    inputVariables = {"authentication", "configuration", "input"},
+    type = "io.camunda:aws-textract:1")
+@ElementTemplate(
+    id = "io.camunda.connectors.AWSTEXTRACT.v1",
+    name = "AWS Textract Outbound Connector",
+    description =
+        "Automatically extract printed text, handwriting, layout elements, and data from any document",
+    inputDataClass = TextractRequest.class,
+    version = 1,
+    propertyGroups = {
+      @ElementTemplate.PropertyGroup(id = "authentication", label = "Authentication"),
+      @ElementTemplate.PropertyGroup(id = "configuration", label = "Configuration"),
+      @ElementTemplate.PropertyGroup(id = "input", label = "Configure input")
+    },
+    documentationRef =
+        "https://docs.camunda.io/docs/next/components/connectors/out-of-the-box-connectors/amazon-textract/",
```

Comment:
Please don't forget to add documentation after the connector is completed. Ping me or @Oleksiivanov for more details.